cmd/primarycmd/addValidator.go

```diff
@@ -0,0 +1,202 @@
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ava-labs/avalanche-cli/cmd/nodecmd"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanchego/genesis"
+
+	"github.com/ava-labs/avalanche-cli/cmd/subnetcmd"
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/subnet"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/spf13/cobra"
+)
+
+var (
+	deployTestnet                bool
+	deployMainnet                bool
+	keyName                      string
+	subnetAuthKeys               []string
+	outputTxPath                 string
+	useLedger                    bool
+	ledgerAddresses              []string
+	nodeIDStr                    string
+	weight                       uint64
+	startTimeStr                 string
+	duration                     time.Duration
+	ErrMutuallyExlusiveKeyLedger = errors.New("--key and --ledger,--ledger-addrs are mutually exclusive")
+	ErrStoredKeyOnMainnet        = errors.New("--key is not available for mainnet operations")
+)
+
+type jsonProofOfPossession struct {
+	PublicKey         string `json:"publicKey"`
+	ProofOfPossession string `json:"proofOfPossession"`
+}
+
+// avalanche subnet deploy
+func newAddValidatorCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "addValidator",
+		Short: "Add a validator to Primary Network",
+		Long: `The primary addValidator command adds a node as a validator 
+in the Primary Network`,
+		SilenceUsage: true,
+		RunE:         addValidator,
+		Args:         cobra.ExactArgs(0),
+	}
+	cmd.Flags().StringVarP(&keyName, "key", "k", "", "select the key to use [fuji deploy only]")
+	cmd.Flags().StringVar(&nodeIDStr, "nodeID", "", "set the NodeID of the validator to add")
+	cmd.Flags().Uint64Var(&weight, "weight", 0, "set the staking weight of the validator to add")
+	cmd.Flags().StringVar(&startTimeStr, "start-time", "", "UTC start time when this validator starts validating, in 'YYYY-MM-DD HH:MM:SS' format")
+	cmd.Flags().DurationVar(&duration, "staking-period", 0, "how long this validator will be staking")
+	cmd.Flags().BoolVar(&deployTestnet, "fuji", false, "join on `fuji` (alias for `testnet`)")
+	cmd.Flags().BoolVar(&deployTestnet, "testnet", false, "join on `testnet` (alias for `fuji`)")
+	cmd.Flags().BoolVar(&deployMainnet, "mainnet", false, "join on `mainnet`")
+	cmd.Flags().StringSliceVar(&subnetAuthKeys, "subnet-auth-keys", nil, "control keys that will be used to authenticate add validator tx")
+	cmd.Flags().StringVar(&outputTxPath, "output-tx-path", "", "file path of the add validator tx")
+	cmd.Flags().BoolVarP(&useLedger, "ledger", "g", false, "use ledger instead of key (always true on mainnet, defaults to false on fuji)")
+	cmd.Flags().StringSliceVar(&ledgerAddresses, "ledger-addrs", []string{}, "use the given ledger addresses")
+	return cmd
+}
+
+func promptProofOfPossession() (jsonProofOfPossession, error) {
```

Comment:
I believe this can return a signer.ProofOfPossesion. Made by:
1) filling the public fields PublicKey and ProofOfPosession of the struct
2) calling Verify() on it to make all data consistent
so no need for jsonProofOfPossession struct